### PR TITLE
Various Ballistic Gun Improvements

### PR DESCRIPTION
**Pull request by RandomColonist**
- Added recoil modifier var to most of the ballistic weapons. Previously, only Gladstone had it, and it was 0.4, making it basically one of the most OP weapons aviable in game. Now, guns assigned recoil modifier according to their caliber, type, size, etc. Peashooter guns, for example, range from 0.2 to 0.4 recoil mod, and AR typically have 1.2. You can look this up in changes.
- Excelsior AK was given autofire mode, three round bursts on both AKs were changed to short bursts. Description changed, lore joke was added, proabably needs approval.
- Z8 Bulldog was remade into carbine from AR, mag capasity was buffed to 20, added zoom, recoil mod, etc.
- Gladstone an Regulator now actually can compete, made Regulator more damaging with standart recoil, and Gladstone less damaging with less recoil and more capacity.
- Added auto-eject to some modern guns. Added sound of it to them also. Added flag which forbids tactical reload for said guns because reasons.
- Zoom was adjusted across the deck. Most guns now have 1.2 zoom which meant for red-dots, mechanical sights and shit like this. Only AMR and Valkyrie now have 2.0 zoom.
- Various naming improvements, unused vars removal, damage multiplier adjustment and much more. Most of it explained in //, so next who will touch guns will have slight idea about why vars was set like this, i hope.
- Due to all changes above and for sake of standartisation of economy, most gun prices are changed. Again, tried to describe most in //.

I hope i covered everything. Energy guns will be in other PR. This is not final version, most likely, as i require input, particularly from @EvenInDeathIStillServe

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/automatic/z8.dm [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
 	matter = list(MATERIAL_PLASTEEL = 20, MATERIAL_PLASTIC = 6)
-	price_tag = 3500
+	price_tag = 3200 //old but gold, decent AP caliber, underbarrel GL, mild recoil and 20-round mags. Better than FS AK.
 	ammo_type = "/obj/item/ammo_casing/a556"
 	fire_sound = 'sound/weapons/guns/fire/batrifle_fire.ogg'
 	slot_flags = SLOT_BACK
 	load_method = MAGAZINE
 	mag_well = MAG_WELL_CIVI_RIFLE
 	magazine_type = /obj/item/ammo_magazine/a556
-	auto_eject = 1
```
Comment (TorinoFermic) [inline code/modules/projectiles/guns/projectile/automatic/z8.dm]: Why you remove this line when the Z8 has auto-ejecting audio with it originally.
Do you plan also remove auto_eject_sound ?
If not, just copy audio line and put with all guns where the auto_eject was left enabled.

Comment (RandomColonist) [reply code/modules/projectiles/guns/projectile/automatic/z8.dm]: Will do, thanks.

File: code/modules/projectiles/guns/projectile/automatic/z8.dm [hunk 2]
```diff
@@ -1,36 +1,34 @@
 /obj/item/weapon/gun/projectile/automatic/z8
-	name = "FS AR 5.56x45mm \"Z8 Bulldog\""
-	desc = "The Z8 Bulldog is an older model bullpup carbine, made by \"Frozen Star\". Uses 5.56mm rounds."
+	name = "FS CAR 5.56x45mm \"Z8 Bulldog\""
+	desc = "The Z8 Bulldog is an older model bullpup carbine, made by \"Frozen Star\". Has underbarrel grenade launcher which are compitable with most modern grenade types. Uses 5.56mm rounds."
```
Comment (EvenInDeathIStillServe) [inline code/modules/projectiles/guns/projectile/automatic/z8.dm]: desc = "The Z8 Bulldog is an older bullpup carbine model, made by \"Frozen Star\". It includes an underbarrel grenade launcher which is compatible with most modern grenade types. Uses 5.56mm rounds."

File: code/modules/projectiles/guns/projectile/revolver/mateba.dm [hunk 3]
```diff
@@ -1,6 +1,9 @@
 /obj/item/weapon/gun/projectile/revolver/mateba
 	name = "FS REV .357 \"Mateba\""
-	desc = "Uses old school .357 ammo."
+	desc = "Old, reliable and robust designed hand-cannon, this is a gun for a real officer, not just self-proclaimed \"leader\". Revolver of choice when you not just need to make a hole in someone, but to tore their limbs off. Uses old school .357 ammo."
```
Comment (EvenInDeathIStillServe) [inline code/modules/projectiles/guns/projectile/revolver/mateba.dm]: desc = "Old, reliable, hand-cannon with a robust design. This is a gun for a real officer, not just a self-proclaimed \"leader\". Revolver of choice when you want to put someone down. Permanently. Uses old school .357 ammo."

**Comment by RandomColonist** [comment]
Added requested changes regardless auto-ejection, shotgun balance and style with grammar.

**Pull request merged**